### ISSUE TITLE
CarPlay Support

### ISSUE THREAD

**Issue by benonymity** [advplyr/audiobookshelf-app#475]
### Is your feature request related to a problem? Please describe.
Feature parity with Android and better app usability when in the car.

### Describe the solution you'd like
An entitlement from Apple will need to be procured in order to develop a CarPlay extension. We'll also need to simply some data models that rely on Realm or Capacitor so that they are more portable--maybe even making a Swift package for the ABS API for easier integration into other apps.

**Comment by advplyr** [comment]
I think I already took the steps for developing for CarPlay. Either way I don't think it will effect development, right? Just deployment to the Apple store

**Comment by benonymity** [comment]
If we have (or write) an OpenAPI v3 schema for ABS, we could use this project to generate an SDK for ABS automatically: https://github.com/CreateAPI/CreateAPI. The native iOS Jellyfin app used that for their API package. Publishing something like that would make also make integration into 3rd party apps like Prologue easier. Something like https://github.com/audiobookshelf-flutter/audiobookshelf_api but in Swift.

**Comment by benonymity** [comment]
See some of the work done here: https://github.com/rasmuslos/audiobooks/tree/master/Shared/Models

**Comment by slyfoxgh** [comment]
+1 Really need CarPlay support.

**Comment by simonaplin** [comment]
+1

**Comment by WWoodpecker** [comment]
+1 Really

**Comment by srvoets** [comment]
+1, might have to go back to pocketcasts because its straight up dangerous using this app in the car.

**Comment by eximo84** [comment]
Agreed. CarPlay is a must have for safety reasons. Especially as android have android auto support.

**Comment by xstefanx** [comment]
Very much would like to see this implemented. Most of my listening time happens in the car and having the safer controls would mean a lot.

**Comment by vdrover** [comment]
Playback via CarPlay is currently "fine". Below are my test results.

Would a GoFundMe help get this feature request moving forward?

1. The app does not show up in the grid of apps in CarPlay. At a minimum, we need the app to show and when selected, be able to browse the library and select a book.
2. Since there is no app icon, the previously used audio app is shown (see Spotify below) in the Carplay dashboard.
3. If you click the book icon in the CarPlay dashboard, you get to the only scene for ABS. Sometimes, this will have an arrow at the top-left. Currently, this arrow has nowhere to go. Ideally, this would go to the ABS app (see 1 above).

![(null) 2023-10-26 at 4 25 10 PM](https://github.com/advplyr/audiobookshelf-app/assets/3143313/9a8edae9-7ff1-4520-857d-1fdbbc21e8af)

![(null) 2023-10-26 at 4 29 36 PM](https://github.com/advplyr/audiobookshelf-app/assets/3143313/abc4c07c-53b7-4a5b-8dea-d3e939c8619d)

**Comment by advplyr** [comment]
I guess that is how iOS handles apps that are playing but have no CarPlay app built

**Comment by woolmonkey** [comment]
@Vdrover Isn't that just the currently playing audio in carplay?  Looks like it.

**Comment by urhovaldeko** [comment]
Yes, that's the "Now Playing" interface. Works the same way on Watch. Unfortunately, "Now Playing" isn't a regular app and won't show up in the recent apps section on the left.

**Comment by TomBayne** [comment]
+1 for this feature.

**Comment by driveridesurf** [comment]
Really enjoying the iOS app. I hope CarPlay integration is something that happens sooner rather than later. Kudos to all the devs working on this project. Thank you.

**Comment by Toomblr** [comment]
+1 for Apple Carplay!

**Comment by nichwall** [comment]
You can show support for the feature by reacting with a thumbs up to the top comment (please don't just comment "+1")
https://www.audiobookshelf.org/faq/#i-have-a-feature-request-how-should-i-bring-this-up

**Comment by fidoriel** [comment]
Maybe this project can borrow some code from https://github.com/BLeeEZ/amperfy. The Projects are license compatible. Imho a very basic last played and start/stop would be more than enough. In this context may be a Siri integration useful (could also be borrowed). More than this should not be done while driving.

**Comment by kitzler-walli** [comment]
I just filed for the CarPlay eligibility - hopefully I'll be able to support the project with a version supporting carplay.

**Comment by sevenlayercookie** [comment]
> I just filed for the CarPlay eligibility - hopefully I'll be able to support the project with a version supporting carplay.

I'm an amateur coder, but I was thinking about forking the app to work on this too. Were you able to file for CarPlay support for your own fork?

**Comment by kitzler-walli** [comment]
I haven't heard from Apple yet - hopefully they will soon decide if I'm worth to develop Apple CarPlay Apps.

**Comment by genebean** [comment]
Thanks for diving into this @kitzler-walli and @sevenlayercookie

**Comment by kitzler-walli** [comment]
I got positive feedback from apple but am currently stuck at 200% workload with almost no spare time. I'll get back to you as soon as I have an update.

**Comment by ivantomica** [comment]
> Hey @kitzler-walli, I understand that it's presumptuous of me to check about the status of CarPlay support given our own daily jobs, but just in case. Has there been any progress? 🙂

@one-harsh , in the meantime, not sure if you already checked it out, but there's quite a good app called "plappa" which has carplay support :-)

**Comment by kitzler-walli** [comment]
Till now I only had a few hours and the facts that this is my first attempt at a carplay app didn't help.

**Comment by kitzler-walli** [comment]
Better late than never but I finally created my apple carplay version - to be honest I did it completely with claude code but it works and I use it in my car currently and am very pleased with the results.

**Comment by fidoriel** [comment]
Do you have a brach @kitzler-walli to try?

**Comment by kitzler-walli** [comment]
I just created a fork and pushed my changes into it: https://github.com/kitzler-walli/audiobookshelf-app
(i had to merge the recent changes of the project - so the current state is untested)

I could also add some users to testflight - by the way - why is the app not yet published in the app store?

**Comment by kitzler-walli** [comment]
Today I verified the current version  - works just fine on my Carplay. (the app does not work in the Carplay simulator, but works just fine on a real device)
